### PR TITLE
hooks: Copy-edits for the Markdown docs (RFC 2119, etc.)

### PR DESCRIPTION
**Pull request by wking**
This is a fairly thorough edit of the hook docs (which are mostly unchanged since #562).  The commits are mostly per-section rewrites, but I'm happy to squash/split/refactor if any particular change is (un)contentious.  Details for each set of changes in the commit messages.

I'd like to rework the hook configuration to more closely match the upstream structure (e.g. we do not expose `timeout`), but I'm leaving that alone for now.  This PR just tightens the docs for the current format.

### CODE REVIEW

File: hooks.md [hunk 1]
```diff
@@ -1,87 +1,54 @@
 # OCI Hooks Configuration
 
-[The OCI Runtime Specification defines POSIX-platform Hooks:](
-https://github.com/opencontainers/runtime-spec/blob/master/config.md#posix-platform-hooks)
+For POSIX platforms, the [OCI runtime configuration][runtime-spec] supports [hooks][spec-hooks] for configuring custom actions related to the life cycle of the container.
+The way you enable the hooks above is by editing the OCI runtime configuration before running the OCI runtime (e.g. [`runc`][runc]).
+CRI-O and `Kpod create` create the OCI do for you, and this documentation allows developers to configure CRI-O to set their intended hooks.
```
Comment (mrunalp) [inline hooks.md]: `..create the OCI do for you..` seems incorrect.

Comment (wking) [reply hooks.md]: > `..create the OCI do for you..` seems incorrect.

Thanks.  Fixed in 60a419868 → cd173dd5e.

**Comment by wking** [comment]
[The integration-test failure][1] seens unrelated.

[1]: https://travis-ci.org/kubernetes-incubator/cri-o/jobs/344076222#L3969

**Comment by rhatdan** [comment]
These look fine to me.

File: hooks.md [hunk 2]
```diff
@@ -1,87 +1,54 @@
 # OCI Hooks Configuration
 
-[The OCI Runtime Specification defines POSIX-platform Hooks:](
-https://github.com/opencontainers/runtime-spec/blob/master/config.md#posix-platform-hooks)
+For POSIX platforms, the [OCI runtime configuration][runtime-spec] supports [hooks][spec-hooks] for configuring custom actions related to the life cycle of the container.
+The way you enable the hooks above is by editing the OCI runtime configuration before running the OCI runtime (e.g. [`runc`][runc]).
+CRI-O and `Kpod create` create the OCI configuration for you, and this documentation allows developers to configure CRI-O to set their intended hooks.
```
[the remark this answers is not in the document]
Comment (wking) [reply hooks.md]: > Could you say podman rather then kpod

`Kpod create` is recycled [from master][1], but I can change it to `podman create`.

[1]: https://github.com/kubernetes-incubator/cri-o/blame/cefb7f8b9e3c305b310b85817e3a2313b077e721/hooks.md#L36

[the remark this answers is not in the document]
Comment (wking) [reply hooks.md]: Updated `Kpod` → `podman` with cd173dd5e → 0b08c8437.

**Pull request merged**